ribbon/contract.py

```diff
@@ -34,10 +35,18 @@ class ContractConnection:
       w3 (object): RPC connection instance
       contract (object): Contract instance
   """
-  def __init__(self, rpc_url: str, rpc_token: str, address: str, abi: dict) -> None:
-    self.address = address
+
+  def __init__(self, address: str, chain: str, abi: dict) -> None:
+    self.chain = chain
+
+    if self.chain not in CHAINS:
```

Comment:
This is related to having a configuration, 
so when we create that configuration we can select which chains we are in,
and not to specify it in here.

We'd have a better API for the user if the error is at the time of creating the configuration.